internal/core/properties/properties_animations.rs

```diff
@@ -71,18 +86,32 @@ impl<T: InterpolatedPropertyValue + Clone> PropertyValueAnimationData<T> {
                 {
                     self.state = AnimationState::Animating { current_iteration };
 
-                    let progress =
-                        (time_progress as f32 / self.details.duration as f32).clamp(0., 1.);
+                    let progress = {
+                        let progress =
+                            (time_progress as f32 / self.details.duration as f32).clamp(0., 1.);
+                        if reversed(current_iteration) {
+                            1. - progress
+                        } else {
+                            progress
+                        }
+                    };
                     let t = crate::animations::easing_curve(&self.details.easing, progress);
                     let val = self.from_value.interpolate(&self.to_value, t);
 
                     (val, false)
                 } else {
-                    self.state = AnimationState::Done;
+                    self.state =
+                        AnimationState::Done { interation_count: (current_iteration - 1).max(0) };
```

Comment:
Whoops, just noticed these are unsigned, so this `max` is useless - I'm pretty sure it's never going to underflow anyway, given we bail out on `iteration_count == 0.`.